src/Protos/Grpc/persistent.proto

```diff
@@ -19,7 +19,12 @@ message ReadReq {
 	}
 
 	message Options {
-		event_store.client.shared.StreamIdentifier stream_identifier = 1;
+		event_store.client.shared.StreamIdentifier stream_identifier = 1 [deprecated = true];
+		oneof stream_option {
```

Comment:
Do we expect to add more options to StreamOptions / AllOptions? If not, then according to https://developers.google.com/protocol-buffers/docs/proto#updating, we could do

```
oneof stream_option {
  event_store.client.shared.StreamIdentifier stream_identifier = 1;
  Empty all = 5;
}
```

> Changing a single optional value into a member of a new oneof is safe and binary compatible. Moving multiple optional fields into a new oneof may be safe if you are sure that no code sets more than one at a time. Moving any fields into an existing oneof is not safe.

Reply:
@thefringeninja thanks, i'll apply this change to `ReadReq` and `DeleteReq` as well :+1:

Reply:
@shaan1337 the changes suggested by @thefringeninja look good to me for `ReadReq` and `DeleteReq`, I don't think we're going to need any more options on StreamOptions/AllOptions

Reply:
thanks @hayley-jean . I've applied the changes.